azure-toolkit-libs/azure-toolkit-auth-lib/src/main/java/com/microsoft/azure/tools/auth/core/maven/AzureContextExecutor.java

```diff
@@ -0,0 +1,49 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for
+ * license information.
+ */
+
+package com.microsoft.azure.tools.auth.core.maven;
+
+import com.microsoft.aad.adal4j.AuthenticationContext;
+import com.microsoft.aad.adal4j.AuthenticationResult;
+import org.apache.commons.lang3.StringUtils;
+
+import java.net.MalformedURLException;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+public class AzureContextExecutor {
+    private String baseUrl;
+    private AcquireTokenFunction acquireTokenFunc;
+
+    public AzureContextExecutor(String baseUrl, AcquireTokenFunction acquireTokenFunc) {
+        if (StringUtils.isBlank(baseUrl)) {
+            throw new IllegalArgumentException("Parameter 'baseUrl' cannot be empty.");
+        }
```

Comment:
how about Preconditions.checkArgument(StringUtils.isNotBlank(baseUrl), "Parameter 'baseUrl' cannot be empty.") ?